### ISSUE TITLE
AddRow causing "maximum number of cell styles was exceeded" error when inserting large number of rows including dates.

### ISSUE THREAD

**Issue by cfprogrammer** [cfsimplicity/lucee-spreadsheet#85]
I am attaching a file that simulates the process I am trying to do.

First there are 3 queries, the first two are of reasonable size and the third is very large (see around line 93) if I have a value of 2000 everything seems fine but if that value increase (normally have in excess of 9000 records in the third query) if throws the 'style' error.

The spreadsheet instantiation is near line 137 

I have pulled the function out of the cfc and included it in the attached file for simplicity.  

Each of the 3 queries are run through the 'monthlySpreadsheetDisplay' function.  This function groups and subtotals each financial class in each profit center. Then at the bottom of each spread sheet it aggregates a bit differently.

After all data has been added to each sheet the formatting is applied (see near line 554) based on row numbers that I have been keeping track of in a couple of arrays.  Even if I comment out all of the formatting. I still get the error :
```

The maximum number of cell styles was exceeded. You can define up to 4000 styles 
in a .xls workbook
Cause 	java.lang.IllegalStateException

```

The detail rows are being added by wrapping them in numberFormat (see near line 295).  If they are not added with that any zero added is added as FALSE not 0.  As you can see in the subtotal of the "H" column.  The detail of that column is being wrapped in numberFormat but the subtotal is not and as the values in that column are 0 the subtotal is also zero.

I hope that this is enough information to help you find a solution.  Thanks again for your incredible library and the amazing responsiveness you have had in helping me.

[monthlySpreadsheet2.zip](https://github.com/cfsimplicity/lucee-spreadsheet/files/362865/monthlySpreadsheet2.zip)

**Comment by cfsimplicity** [comment]
Thanks for putting the detailed sample together. I'll see if I can address this.

**Comment by cfsimplicity** [comment]
Turns out it's not the formatted numbers in your data that are causing the problem, but the dates, each of which causes a new style to be created.

Fix coming up in 0.7.6 which uses a more efficient way of applying the date style.  Seems to work fine with your code.

**Comment by cfprogrammer** [comment]
Thank you so much, works great
